ccr/display.php

```diff
@@ -1,68 +1,68 @@
 <?php
 
+/**
+ * display.php  Is responsible for display a CCR/CCD/CCDA document previewed from the documents folder.
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Ajil P.M <ajilpm@zhservices.com>
+ * @author    Brady Miller <brady.g.miller@gmail.com>
+ * @author    Stephen Nielson <snielson@discoverandchange.com>
+ * @copyright Copyright (c) 2011 Z&H Consultancy Services Private Limited <ajilpm@zhservices.com>
+ * @copyright Copyright (c) 2013 Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2022 Discover and Change <snielson@discoverandchange.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
 require_once(dirname(__FILE__) . "/../interface/globals.php");
 
+use OpenEMR\Events\PatientDocuments\PatientDocumentViewCCDAEvent;
+use OpenEMR\Common\Twig\TwigContainer;
+use OpenEMR\Common\Logging\SystemLogger;
+
 $type = $_GET['type'];
 $document_id = $_GET['doc_id'];
 $d = new Document($document_id);
-$url =  $d->get_url();
-$storagemethod = $d->get_storagemethod();
-$couch_docid = $d->get_couch_docid();
-$couch_revid = $d->get_couch_revid();
 
-if ($couch_docid && $couch_revid) {
-    $couch = new CouchDB();
-    $resp = $couch->retrieve_doc($couch_docid);
-    $xml = base64_decode($resp->data);
-    if ($content == '' && $GLOBALS['couchdb_log'] == 1) {
-        $log_content = date('Y-m-d H:i:s') . " ==> Retrieving document\r\n";
-        $log_content = date('Y-m-d H:i:s') . " ==> URL: " . $url . "\r\n";
-        $log_content .= date('Y-m-d H:i:s') . " ==> CouchDB Document Id: " . $couch_docid . "\r\n";
-        $log_content .= date('Y-m-d H:i:s') . " ==> CouchDB Revision Id: " . $couch_revid . "\r\n";
-        $log_content .= date('Y-m-d H:i:s') . " ==> Failed to fetch document content from CouchDB.\r\n";
-        //$log_content .= date('Y-m-d H:i:s')." ==> Will try to download file from HardDisk if exists.\r\n\r\n";
-        $this->document_upload_download_log($d->get_foreign_id(), $log_content);
-        die(xlt("File retrieval from CouchDB failed"));
+
+try {
+    $twig = new TwigContainer(null, $GLOBALS['kernel']);
+    if (!$d->can_access($_SESSION['user_id'])) {
```

Comment:
check out this PR (brought into codebase about a week ago) where clarified how to use the can_access function:
https://github.com/openemr/openemr/pull/5235

(basically just don't include a parameter and then the acl inner functions will default to $_SESSION['authUser'] )